.github/actions/configure-aws-credentials/action.yml

```diff
@@ -0,0 +1,59 @@
+name: 'Configure AWS Credentials'
+description: 'Configure AWS Credentials for a given application and |
+  environment so that the GitHub Actions workflow can access AWS resources. |
+  This is wrapper around https://github.com/aws-actions/configure-aws-credentials |
+  that first determines the account, role, and region based on the |
+  account_names_by_environment configuration in app-config'
+inputs:
+  app_name:
+    description: 'Name of application folder under /infra'
+    required: true
+  environment:
+    description: 'Name of environment (dev, staging, prod) that AWS resources live in, or "shared" for resources that are shared across environments'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Get AWS account authentication details (AWS account, IAM role, AWS region)
+      run: |
+        # Get AWS account authentication details (AWS account, IAM role, AWS region)
+        # associated with the application environment to figure out which AWS
+        # account to log into, which IAM role to assume, and which AWS region to use
+
+        echo "::group::AWS account authentication details"
+
+        terraform -chdir=infra/project-config init > /dev/null
+        terraform -chdir=infra/project-config refresh > /dev/null
+        AWS_REGION=$(terraform -chdir=infra/project-config output -raw default_region)
+        echo "AWS_REGION=$AWS_REGION"
+        GITHUB_ACTIONS_ROLE_NAME=$(terraform -chdir=infra/project-config output -raw github_actions_role_name)
+        echo "GITHUB_ACTIONS_ROLE_NAME=$GITHUB_ACTIONS_ROLE_NAME"
+
+        terraform -chdir=infra/${{ inputs.app_name }}/app-config init > /dev/null
+        terraform -chdir=infra/${{ inputs.app_name }}/app-config refresh > /dev/null 
+        ACCOUNT_NAME=$(terraform -chdir=infra/${{ inputs.app_name }}/app-config output -json account_names_by_environment | jq -r .${{ inputs.environment }})
+        echo "ACCOUNT_NAME=$ACCOUNT_NAME"
+
+        # Get the account id associated with the account name
+        # by listing files in the infra/accounts folder,
+        # filtering to files that start with the account name followed by a period,
+        # filtering to files that end in s3.tfbackend
+        # (there should be one file that looks like <ACCOUNT_NAME>.<ACCOUNT_ID>.s3.tfbackend),
+        # and then stripping the suffix and prefix to just end up with <ACCOUNT_ID>.
+        ACCOUNT_ID=$(ls -1 infra/accounts | grep "$ACCOUNT_NAME." | grep s3.tfbackend | sed 's/.s3.tfbackend//' | sed "s/$ACCOUNT_NAME.//")
```

Comment:
I think this can be simplified using shell glob and cut instead of grep and sed:

```suggestion
        ACCOUNT_ID=$(ls $ACCOUNT_NAME.*.s3.tfbackend | cut --delimiter=. --fields=2)
```

Testing locally with an example file:
![Screenshot_2023-08-04_12-28-58](https://github.com/navapbc/template-infra/assets/3811269/334986bb-9b59-4721-b2ac-2be3199ade36)